core/cthreads.h

```diff
@@ -0,0 +1,91 @@
+#ifndef CTHREADS_H
+#define CTHREADS_H
+
+struct cthreads_args {
+  void *(*func)(void *data);
+  void *data;
+};
+
+#if __unix__
+#include <pthread.h>
+#elif _WIN32
+#include <windows.h>
+#endif
+
+struct cthreads_thread {
+    #ifdef __unix__
+    pthread_t pThread;
+    #elif _WIN32
+    HANDLE wThread;
+    #endif
+};
+
+struct cthreads_mutex {
+    #ifdef __unix__
+    pthread_mutex_t pMutex;
+    #elif _WIN32
+    HANDLE wMutex;
+    #endif
+};
+
+struct cthreads_cond {
+    #ifdef __unix__
+    pthread_cond_t pCond;
+    #elif _WIN32
+    HANDLE wCond;
+    #endif
+};
+
+struct cthreads_rwlock {
+    #ifdef __unix__
+    pthread_rwlock_t pRWLock;
+    #elif _WIN32
+    HANDLE wRWLock;
+    #endif
+};
+
+int cthreads_thread_create(struct cthreads_thread *thread, void *(*func)(void *data), void *data);
+
+int cthreads_thread_detach(struct cthreads_thread *thread);
+
+void cthreads_thread_close(void *code);
+
+struct cthreads_mutex __cthreads_mutex_init();
+
+int cthreads_mutex_init(struct cthreads_mutex *mutex);
+
+int cthreads_mutex_lock(struct cthreads_mutex *mutex);
+
+int cthreads_mutex_trylock(struct cthreads_mutex *mutex);
+
+int cthreads_mutex_unlock(struct cthreads_mutex *mutex);
+
+void cthreads_mutex_destroy(struct cthreads_mutex *mutex);
+
+int cthreads_cond_init(struct cthreads_cond *cond);
+
+int cthreads_cond_signal(struct cthreads_cond *cond);
+
+int cthreads_cond_broadcast(struct cthreads_cond *cond);
+
+int cthreads_cond_destroy(struct cthreads_cond *cond);
+
+int cthreads_cond_wait(struct cthreads_cond *cond, struct cthreads_mutex *mutex);
+
+int cthreads_wait_close_thread(struct cthreads_thread *thread, void *code);
+
+int cthreads_rwlock_init(struct cthreads_rwlock *rwlock);
+
+int cthreads_rwlock_rdlock(struct cthreads_rwlock *rwlock);
+
+int cthreads_rwlock_unlock(struct cthreads_rwlock *rwlock);
+
+int cthreads_rwlock_wrlock(struct cthreads_rwlock *rwlock);
+
+int cthreads_rwlock_destroy(struct cthreads_rwlock *rwlock);
+
+int cthreads_equal(struct cthreads_thread thread1, struct cthreads_thread thread2);
+
+struct cthreads_thread cthreads_self();
+
+#endif /* CTHREADS_H */
```

Comment:
I know this is nitpicky, but, I'd cut those extra newlines between the function definition, or fill 'em in with some Doxygen material explaining how to use CThreads. Also -- make sure you've got that newline at the end of everything to shut the compiler up.